tests/acceptance/bootstrap/WebDav.php

```diff
@@ -1573,32 +1573,34 @@ public function checkElementList(
 				//       /some-folder%20with%20spaces/sub-folder
 				// So we need both $elementToRequest and $expectedElement
 				$expectedElement = $this->encodePath($elementToRequest);
+
 				$responseXmlObject = $this->listFolderAndReturnResponseXml(
 					$user,
 					$elementToRequest,
 					"1"
 				);
 
-				// TODO: make it work for folder entries
-				// Doesn't work for folder entries
-				// as the folder entry has trailing '/' in d:href
 				$webdavPath = "/" . $this->getFullDavFilesPath($user) . $expectedElement;
+				// return xmlobject that matches the x-path pattern.
 				$element = $responseXmlObject->xpath(
-					"//d:response/d:href[text() = \"$webdavPath\"]"
+					"//d:response/d:href"
```

Comment:
this will return multiple hrefs. but we want to get the specific one

Reply:
```suggestion
					"//d:response/d:href[text() = \"$webdavPath\" or text() = \"{$webdavPath}/\"]"
```

Reply:
That is why `if (!$expectedToBeListed && isset($element[0])` is done. The first element is always the current one

Reply:
Then it is good to add the comment just above it mentioning why only get first element.
And this would be good to do rather than `$element[0]`:
```php
$element = $responseXmlObject->xpath("//d:response/d:href")[0];
```

Reply:
This doesn't work because there are some negative tests. https://github.com/owncloud/ocis/blob/00e51d58b4460a1d33455a390f843c1792733d9b/tests/acceptance/features/coreApiMain/checksums.feature#L171-L175

Reply:
:+1: But I still need a comment there to clarify that we will only check the first element because the requested resource will always be the first one